pages/agent/v3/cli_artifact.md

```diff
@@ -289,3 +290,76 @@ up, see our [Google Cloud installation guide](/docs/agent/v3/gcloud#uploading-ar
 You can configure the `buildkite-agent artifact` command to store artifacts in
 Artifactory. For instructions for how to set this up, see our
 [Artifactory guide](/docs/integrations/artifactory).
+
+## Using your private Azure Blob container
+
+>📘 Support for uploading artifacts to Azure Blob Storage was added in Agent
+> v3.53.0.
+
+You can configure the `buildkite-agent artifact` command to store artifacts in
+your private [Azure Blob Storage container](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction).
+
+### Preparation
+
+Firstly, make sure that each agent has access to Azure credentials.
+[By default](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#readme-defaultazurecredential)
+these can be provided using Azure environment variables such as
+`AZURE_CLIENT_ID`, loaded by a Kubernetes workload identity hook, loaded on a
+host with Azure Managed Identity enabled, or loaded from a user logged in with
+the Azure CLI.
+
+You can also use an account key or connection string by setting one of these
+environment variables:
+
+```shell
+# To use an account key:
+export BUILDKITE_AZURE_BLOB_ACCESS_KEY='...'
+
+# To use a connection string:
+export BUILDKITE_AZURE_BLOB_CONNECTION_STRING='...'
+```
+
+Since these can contain access credentials, they are redacted from job logs by
+default. To override this behaviour set the `BUILDKITE_REDACTED_VARS`
```

Comment:
I don't think it's necessary to say how to unredact the secret values here? Maybe just link to the config option.